learn/sherif/main_test/src/main/java/frc/robot/subsystems/IntakeControl.java

```diff
@@ -7,44 +7,44 @@
 import frc.robot.Constants;
 
 public class IntakeControl extends SubsystemBase {
-    public final WPI_TalonSRX m_throat_left;
-    public final WPI_TalonSRX m_throat_right;
-    public final WPI_TalonSRX m_elevator_left;
-    public final WPI_TalonSRX m_elevator_right;
-    public final WPI_TalonSRX m_feeder;
-    private double m_throat_speed = 0.0;
-    private double m_elevator_speed = 0.0;
-    private double m_feeder_speed = 0.0;
-    private final Solenoid m_solenoid = null;
+    public final WPI_TalonSRX throatLeft;
+    public final WPI_TalonSRX throatRight;
+    public final WPI_TalonSRX elevatorLeft;
+    public final WPI_TalonSRX elevatorRight;
+    public final WPI_TalonSRX feeder;
+    private double throatSpeed = 0.0;
+    private double elevatorSpeed = 0.0;
+    private double feederSpeed = 0.0;
+    private final Solenoid solenoid = null;
```

Comment:
We don't actually use the Solenoid class. A Solenoid object essentially controls an on/off switch, which is why it's enabled/disabled with a boolean value. Either air is let in or it's not. To use an air cylinder, we would need two solenoids, one to control air to the "front" and one to control air to the "back". Actuating a cylinder would mean letting compressed air into one side and opening the other side out to the atmosphere. There are two ways of doing this. One would be to use two Solenoid objects and control them both separately, always giving them opposite boolean inputs. The other is to use the DoubleSolenoid class, which automatically operates the two solenoids, and takes predefined enums as inputs. DoubleSolenoids are used like so:

```
import edu.wpi.first.wpilibj.DoubleSolenoid;
import edu.wpi.first.wpilibj.DoubleSolenoid.Value;

private final DoubleSolenoid m_intakeSolenoid = new DoubleSolenoid(0, 1); // these would be values found in Constants.java
/**
 * Lowers the intake arm. (Extends the cylinder)
 */
public void lowerIntake() {
    intakeSolenoid.set(Value.kForward);
}

/**
 * Raises the intake arm. (Retracts the cylinder)
 */
public void raiseIntake() {
    intakeSolenoid.set(Value.kReverse);
}
```

[Wiki page](https://docs.wpilib.org/en/stable/docs/software/actuators/pneumatics.html?highlight=DoubleSolenoid#double-solenoids-in-wpilib)
[API page](https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/DoubleSolenoid.html)

Reply:
I just left the solenoid code there and left it there, I'll keep this in notice and update this for the robot when it comes around probably pretty soon